code-studio/src/js/callouts.js

```diff
@@ -112,16 +124,49 @@ module.exports = function createCallouts(callouts) {
     }
 
     if (callout.on) {
-      $(window).on(callout.on, function () {
-        if (!callout.seen && $(selector).length > 0) {
+      $(window).on(callout.on, function (e, action) {
+        if (!config.codeStudio.selector) {
+          config.codeStudio.selector = selector;
+        }
+        var lastSelector = config.codeStudio.selector;
+        $(window).trigger('prepareforcallout', [config.codeStudio]);
+        if (lastSelector !== config.codeStudio.selector && $(lastSelector).length > 0) {
+          $(lastSelector).qtip(config).qtip('destroy');
+        }
+        if ($(config.codeStudio.selector).length > 0) {
+          if (action === 'hashchange' || action === 'hashinit' || !callout.seen) {
+            $(config.codeStudio.selector).qtip(config).qtip('show');
+          }
           callout.seen = true;
-          $(selector).qtip(config).qtip('show');
         }
       });
-    } else {
+    } else if (!callout.seen) {
       $(selector).qtip(config).qtip('show');
     }
   });
+
+  // Insert a hashchange handler to detect triggercallout= hashes and fire
+  // appropriate events to open the callout
+  function detectTriggerCalloutOnHash(event) {
```

Comment:
There isn't one now, but if someone wanted to point directly to a level URL with the anchor tag (to call attention to something via a callout at load time), it seemed like we should support it and not ignore it, so that it is a more officially supported part of the URL format.

> On Jun 9, 2016, at 2:29 PM, Brent Van Minnen <notifications@github.com> wrote:
> 
> In code-studio/src/js/callouts.js:
> 
> >        $(selector).qtip(config).qtip('show');
> >      }
> >    });
> > +
> > +  // Insert a hashchange handler to detect triggercallout= hashes and fire
> > +  // appropriate events to open the callout
> > +  function detectTriggerCalloutOnHash(event) {
> What is the scenario where the hash will be there on page load?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>